extensions/blocks/social-previews/upgrade.js

```diff
@@ -43,10 +59,79 @@ const SocialPreviewsUpgrade = function SocialPreviewsUpgrade() {
 						) }
 					</li>
 				</ul>
-				<Button isPrimary>{ __( 'Upgrade', 'jetpack' ) }</Button>
+
+				<Button
+					href={ href } // Only for server-side rendering, since onClick doesn't work there.
+					isPrimary
+					label={ __( 'Purchase a business plan to access social previews', 'jetpack' ) }
+					onClick={ autosaveAndRedirect }
+					target="_top"
+				>
+					{ __( 'Upgrade', 'jetpack' ) }
+				</Button>
 			</div>
 		</div>
 	);
 };
 
-export default SocialPreviewsUpgrade;
+export default compose( [
+	withSelect( select => {
+		const plan = select( 'wordpress-com/plans' ).getPlan( BUSINESS_PLAN );
+		const postId = select( 'core/editor' ).getCurrentPostId();
+		const postType = select( 'core/editor' ).getCurrentPostType();
+
+		const planPathSlug = plan?.path_slug;
+		const domain = getSiteFragment();
+
+		// The editor for CPTs has an `edit/` route fragment prefixed.
+		const postTypeEditorRoutePrefix = [ 'page', 'post' ].includes( postType ) ? '' : 'edit';
+
+		// Post-checkout: redirect back here.
+		const redirect_to = isSimpleSite()
+			? addQueryArgs(
+					'/' + compact( [ postTypeEditorRoutePrefix, postType, domain, postId ] ).join( '/' ),
+					{
+						plan_upgraded: 1,
+					}
+			  )
+			: addQueryArgs(
+					window.location.protocol + `//${ domain.replace( '::', '/' ) }/wp-admin/post.php`,
+					{
+						action: 'edit',
+						post: postId,
+						plan_upgraded: 1,
+					}
+			  );
+
+		const href =
+			planPathSlug &&
+			addQueryArgs( `https://wordpress.com/checkout/${ domain }/${ planPathSlug }`, {
+				redirect_to,
+			} );
```

Comment:
```suggestion
			addQueryArgs( `https://wordpress.com/checkout/${ domain }/${ planPathSlug }`, {
				redirect_to: isSimpleSite()
			} );
```

As discussed before the `redirect_to` var doesn't follow JS variable naming rules. I now understand this is because it is required as it forms part of a URL query args.

That said, given that this is only now used in one location that I can see, we can probably just inline the var directly where it's used.